lib/puppet/functions/simp_pki_service/validate_ca_hash.rb

```diff
@@ -0,0 +1,86 @@
+# Validate that the passed Hash of CAs meets the following requirements:
+#
+# * There is at least one Root CA
+# * Each Sub CA has defined a parent CA
+# * Each Sub CA Has a Root CA in the list
+#
+# This does **not** check that the options are valid, that is left to the
+# defined types to which the hash components are passed.
+#
+Puppet::Functions.create_function(:'simp_pki_service::validate_ca_hash') do
+
+  # @param ca_hash The CA Hash to process
```

Comment:
I realize the examples help clarify the undocumented CA hash structure, but I would still like to see *some* description of the CA hash structure, here and in init.pp.
* Each entry is a hash containing parameters specific to the CA  named in the key
* Parameters to be inspected by this function are 'root_ca', and 'parent_ca'
* There can be 1 or more Root CAs
* There can be 0 or more Sub-CAs
* There cannot be any Sub-Sub-CAs (should it? Can the user to provide a multi-level chain of CAs in $simp_pki_service::custom_cas?)